lib/streamlit/connections/snowpark_connection.py

```diff
@@ -174,6 +195,16 @@ def session(self) -> Iterator["Session"]:
 
         Information on how to use Snowpark sessions can be found in the
         [Snowpark documentation](https://docs.snowflake.com/en/developer-guide/snowpark/python/working-with-dataframes).
+
+        Example
+        -------
+        >>> import streamlit as st
+        >>>
+        >>> conn = st.experimental_connection("snowpark")
+        >>> with conn.session() as session:
+        ...     df = session.table("mytable").limit(10).to_pandas()
+        ...
+        >>> st.dataframe(df)
```

Comment:
Heads up that this will be invalid now that we're making `session` a property and `safe_session` the contextmanager method. Should be something that will just come up as a merge conflict, though, so can be fixed then